integration/spark/shared/src/main/java/io/openlineage/spark/agent/lifecycle/plan/BigQueryNodeVisitor.java

```diff
@@ -84,13 +88,40 @@ private Optional<Supplier<BigQueryRelation>> bigQuerySupplier(LogicalPlan plan)
   @Override
   public List<D> apply(LogicalPlan x) {
     return bigQuerySupplier(x)
+        .map(s -> s.get())
+        .filter(relation -> getBigQueryTableName(relation).isPresent())
         .map(
-            s -> {
-              BigQueryRelation relation = s.get();
-              String name = relation.tableName();
-              return Collections.singletonList(
-                  factory.getDataset(name, BIGQUERY_NAMESPACE, relation.schema()));
-            })
+            relation ->
+                Collections.singletonList(
+                    factory.getDataset(
+                        getBigQueryTableName(relation).get(),
+                        BIGQUERY_NAMESPACE,
+                        relation.schema())))
         .orElse(null);
   }
+
+  /**
+   * Versions of spark-bigquery-connector differ in implementation of {@link BigQueryRelation}
+   * class. Previously `tableName` method was available in Scala class, recent java implementation
+   * contains `getTableName` instead. This method retrieves the table name regardless of connector
+   * version.
+   *
+   * @return
+   */
+  private Optional<String> getBigQueryTableName(BigQueryRelation relation) {
```

Comment:
This is the main change of the PR: call both methods `tableName` and `getTableName` on `BigQueryRelation` to retrieve name regardless of version.